python/pyarrow/tests/test_ipc.py

```diff
@@ -1153,9 +1153,17 @@ def make_batches():
     batches = iter(UserList(make_batches()))  # weakrefable
     wr = weakref.ref(batches)
 
-    with pa.ipc.RecordBatchReader.from_batches(make_schema(),
-                                               batches) as reader:
+    with pa.RecordBatchReader.from_batches(make_schema(),
+                                           batches) as reader:
         batches = None
         assert wr() is not None
         assert list(reader) == make_batches()
         assert wr() is None
+
+    # ensure we get proper error when not passing a schema
+    # (https://issues.apache.org/jira/browse/ARROW-18229)
+    batches = make_batches()
+    with pytest.raises(TypeError):
+        reader = pa.RecordBatchReader.from_batches(
+            [('field', pa.int64())], batches)
```

Comment:
TBH, it would be nice if the list of fields was implicitly converted into a schema here :-)

Reply:
Yes, I also mentioned this on the JIRA, but we don't do that anywhere else right now where we accept a schema (I think), so we can open a new JIRA to improve that in general